engine/src/flutter/lib/web_ui/test/engine/semantics/semantics_test.dart

```diff
@@ -1122,114 +1041,16 @@ void _testContainer() {
       owner().updateSemantics(builder.build());
       expectSemanticsTree(owner(), '''
 <sem>
-  <sem-c>
     <sem style="z-index: 2"></sem>
     <sem style="z-index: 4"></sem>
     <sem style="z-index: 1"></sem>
     <sem style="z-index: 3"></sem>
-  </sem-c>
 </sem>''');
     }
 
     semantics().semanticsEnabled = false;
   });
 
-  test('container nodes are transparent and leaf children are opaque hit-test wise', () async {
```

Comment:
I think we still need this test. The "container nodes" in the description does not refer to the intermediate container elements that you are removing. It refers to `SemanticsNode` whose only purpose is to contain children and affect their transforms. We want those nodes to be transparent hit-test wise, and we should continue testing for that.